src/lib/OpenEXR/ImfInputFile.cpp

```diff
@@ -338,7 +339,8 @@ bufferedReadPixels (InputFile::Data* ifd, int scanLine1, int scanLine2)
                     switch (toSlice.type)
                     {
                         case UINT: {
-                            unsigned int fill = toSlice.fillValue;
+                            assert(toSlice.fillValue >= 0);
```

Comment:
In this context, an exception would be more appropriate than an assert. Unlikely as it is that fillValue < 0, better to report it as an error than to terminate the process.

Reply:
Ok. I guess I was mainly thinking of setups where `assert` does nothing in "Release" builds, and therefore incurs no runtime cost.

If you feel like it's super unlikely, then I can also just revert the `assert`? (But keep the typecast that fixes the compiler warning, which anyway is the main point of this PR.)

Reply:
True, the use of assert is a little inconsistent, some parts of the code use it, others don't. Since no other parts of the InputFile class use it, might as well not start, especially since as you say, the point of the PR is to address the type conversion. If we're going to introduce data validation, better to do that separately.  So just remove the assert and it looks good to me. Thanks!

Reply:
Dropped the assertion.